### ISSUE TITLE
compact

### ISSUE THREAD

**Issue by kootenpv** [kriadmin/30-seconds-of-python-code#4]
Instead of:

```python
def compact(arr):
    return list(filter(lambda x: bool(x), arr))
```

https://github.com/kriadmin/30-seconds-of-python-code#compact

You can use:

```python
def compact(arr):
    return list(filter(None, arr))
```

since:

    >>> list(filter(None, [0, 1, 2, 3]))
    [1, 2, 3]

**Comment by gerardlouw** [comment]
What about the more obvious
`[x for x in arr if x]`

**Comment by kootenpv** [comment]
@gerardlouw indeed, just thought to stay in the style of filter, but that's a wrong thought :)

**Comment by kriadmin** [comment]
@gerardlouw Ok will one of u like to submit a PR? If not you can tell me so I will update it.

**P.S.** Do not change the readme change the snippet's file in the snippets folder

**Comment by godaygo** [comment]
@kriadmin  I think you should stay with a style of filter but with a slight modification:
```python
def compact(arr):
    return list(filter(bool, arr))
```
The `lambda` function in `list(filter(lambda x: bool(x), arr))` is redundant and should be replaced with `bool`. There are two main reasons to stay with this code: it is a good code pattern and it is the fastest way to achieve the desired result. See this SO answer https://stackoverflow.com/a/3845453/6793085.

Why you should use  `list(filter(bool, arr))` instead of `list(filter(None, arr))`. Because they are equivalent underneath. The only minimal time difference between these snippets occurs from the fact that the bytecode for `bool` would be `LOAD_NAME` and  `LOAD_CONST`  for `None`  which is slightly faster, but this bytecode will be executed only once, so you will see some difference only on a very short sequences.

**Comment by kriadmin** [comment]
Fixed by #23
